doc/user_guide/axes.rst

```diff
@@ -0,0 +1,649 @@
+* ``low_value`` (int) and ``high_value`` (int) are the smallest and largest
+  value.
+* The ``axis`` array stores the values of the axis points. However,
+  depending on the type of axis, this array may be updated from the **defining
+  attributes** as discussed in the following section.
+
+
+.. _Axes_types:
+
+Types of data axes
+------------------
+
+HyperSpy supports different *data axis types*, which differ in how the axis is
+defined: 
+
+* :py:class:`~.axes.DataAxis` defined by an array ``axis``, 
+* :py:class:`~.axes.FunctionalDataAxis` defined by a function ``expression`` or 
+* :py:class:`~.axes.UniformDataAxis` defined by the initial value ``offset``
+  and spacing ``scale``.
+
+The main disambiguation is whether the
+axis is **uniform**, where the data points are equidistantly spaced, or
+**non-uniform**, where the spacing may vary. The latter can become important
+when, e.g., a spectrum recorded over a *wavelength* axis is converted to a
+*wavenumber* or *energy* scale, where the conversion is based on a ``1/x``
+dependence so that the axis spacing of the new axis varies along the length
+of the axis. Whether an axis is uniform or not can be queried through the 
+property ``is_uniform`` (bool) of the axis.
+
+Every axis of a signal object may be of a different type. For example, it is
+common that the *navigation* axes would be *uniform*, while the *signal* axes
+are *non-uniform*.
+
+When an axis is created, the type is automatically determined by the attributes
+passed to the generator. The three different axis types are summarized in the
+following table.
+
+.. table:: BaseDataAxis subclasses.
+
+    +-------------------------------------------------------------------+------------------------+-------------+
+    |                   BaseDataAxis subclass                           |  defining attributes   |  is_uniform |
+    +===================================================================+========================+=============+
+    |                :py:class:`~.axes.DataAxis`                        |         axis           |  False      |
+    +-------------------------------------------------------------------+------------------------+-------------+
+    |           :py:class:`~.axes.FunctionalDataAxis`                   |      expression        |  False      |
+    +-------------------------------------------------------------------+------------------------+-------------+
+    |             :py:class:`~.axes.UniformDataAxis`                    |    offset, scale       |  True       |
+    +-------------------------------------------------------------------+------------------------+-------------+    
+
+.. NOTE::
+
+    Not all features are implemented for non-uniform axes.
+
+
+Uniform data axis
+^^^^^^^^^^^^^^^^^
+
+The most common case is the :py:class:`~.axes.UniformDataAxis`. Here, the axis
+is defined by the ``offset`` and ``scale`` parameters, which determine the
```

Comment:
I believe that the axis is also "defined by" its size, in addition to offset and scale.
I'd like to emphasize in the docs that the size is a required parameter when instantiating an axis object.

Reply:
see #2693